compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/PositionUtil.java

```diff
@@ -81,4 +81,19 @@ static boolean withinRange(LineRange specifiedRange, Location nodePosition) {
 
         return specifiedEndColumn <= nodeEndColumn;
     }
+
+    //  todo to be removed once https://github.com/ballerina-platform/ballerina-lang/discussions/28983 is sorted
+    static boolean withinRightInclusive(LinePosition cursorPos, Location symbolPosition) {
```

Comment:
Shall we rename this to `nodePosition`? `symbolPosition` would mean something slightly different from node position.